spec/total_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'total'
+
+describe Total do
+  describe "#bill" do
+    # can't think of a better way to test this
+    it "should sum up the total cost of the order" do
+      @total = Total.new
+      allow(@total).to receive(:selection).and_return(

```

Comment:
So here we want to test the behaviour of the class `Total` itself, so we don't want to mock it (it's not a dependency of our subject, but the test subject itself). Therefore the whole `allow(...)` stubbing is not needed, we should instead call methods or attributes of `Total` directly to setup the necessary state/context (= the selection hash)